src/braket/simulator_v2/base_simulator_v2.py

```diff
@@ -44,46 +93,48 @@ def run_jaqcd(
                 as a result type when shots=0. Or, if StateVector and Amplitude result types
                 are requested when shots>0.
         """
-        self._validate_ir_results_compatibility(
-            circuit_ir.results,
-            device_action_type=DeviceActionType.JAQCD,
-        )
-        self._validate_ir_instructions_compatibility(
-            circuit_ir,
-            device_action_type=DeviceActionType.JAQCD,
-        )
-        BaseLocalSimulator._validate_shots_and_ir_results(
-            shots, circuit_ir.results, qubit_count
-        )
-
-        operations = [
-            from_braket_instruction(instruction)
-            for instruction in circuit_ir.instructions
-        ]
-
-        if shots and circuit_ir.basis_rotation_instructions:
-            for instruction in circuit_ir.basis_rotation_instructions:
-                operations.append(from_braket_instruction(instruction))
-
-        BaseLocalSimulator._validate_operation_qubits(operations)
-
-        if not shots and circuit_ir.results:
-            result_types = BaseLocalSimulator._translate_result_types(
-                circuit_ir.results
+        if qubit_count is not None:
+            warnings.warn(
+                f"qubit_count is deprecated for {type(self).__name__} and can be set to None"
             )
-            BaseLocalSimulator._validate_result_types_qubits_exist(
-                [
-                    result_type
-                    for result_type in result_types
-                    if isinstance(result_type, TargetedResultType)
-                ],
-                qubit_count,
-            )
-        r = jl.simulate(self._device, [circuit_ir], qubit_count, shots)
+        translated_ir, qubit_count = self._jaqcd_to_jl(circuit_ir, shots)
+        try:
+            r = jl.simulate(self._device, translated_ir, qubit_count, shots)
```

Comment:
Does the underlying Julia simulator need `qubit_count`? If not, can we remove this parameter (and remove the `qubit_count` return altogether from `_jaqcd_to_jl`.

Reply:
It doesn't strictly need it but it does help not having to recompute it